src/main.h

```diff
@@ -160,8 +160,13 @@ extern BlockMap mapBlockIndex;
 extern std::optional<uint64_t> last_block_num_txs;
 extern std::optional<uint64_t> last_block_size;
 extern const std::string strMessageMagic;
-extern CWaitableCriticalSection csBestBlock;
-extern CConditionVariable cvBlockChange;
+
+// These prevent lock-ordering problems in getblocktemplate() RPC
```

Comment:
Please also describe how these variables are intended to interact with one another.